packit_service/models.py

```diff
@@ -1911,3 +1911,83 @@ def to_dict(self):
 
     def __repr__(self):
         return f"GithubInstallationModel(id={self.id}, account={self.account_login})"
+
+
+class SrcGitPRDistGitPRModel(Base):
+    __tablename__ = "src_git_pr_dist_git_pr"
+    id = Column(Integer, primary_key=True)  # our database PK
+    src_git_pull_request_id = Column(Integer, ForeignKey("pull_requests.id"))
+    dist_git_pull_request_id = Column(Integer, ForeignKey("pull_requests.id"))
+    src_git_pull_request = relationship(
+        "PullRequestModel",
+        primaryjoin="SrcGitPRDistGitPRModel.src_git_pull_request_id==PullRequestModel.id",
+        uselist=False,
```

Comment:
Interesting. I haven't seen this syntax...

I see we are not adding the reverse way of the relation -- (? @jpopelka) is it ok to search through all of the entries when we need to find a matching PR for the given PR?

Reply:
> Interesting. I haven't seen this syntax...

Yes, I struggled a little to find a way to have the two columns inside the same table reference the same keys... it seems to me that in this way it is working but I am not sure at all.

> I see we are not adding the reverse way of the relation -- (? @jpopelka) is it ok to search through all of the entries when we need to find a matching PR for the given PR?

Yes maybe I am missing something more, probably I can't figure it out which query you have in mind.

Reply:
Depends on what we want..;)

* With the current version, we can create the mapping model in a single step (when we have both pull requests), but when we have just one and want to get the other (what I hope is the use-case we will need), we need to go through all the mapping models to get it.
* We can also create the relation in a way, that the relation to the mapping model is saved in the pull request so it can be found in one step. But the question is if it is really an issue. (From the database point of view, the current version does not contain circular dependencies so it looks cleaner.)

Reply:
Ok, thanks, I think I got it! I can try to do that in another commit and then we can choose to take it or give it away.

Reply:
Ok, for me it is the same.
If also @lachmanfrantisek agrees then I will remove the last commit.

Reply:
Also [indexing](https://docs.sqlalchemy.org/en/14/core/constraints.html#sqlalchemy.schema.Index) the `dist_git_pull_request_id` and `src_git_pull_request_id` should speed the queries a lot.
Based on [this](https://stackoverflow.com/questions/107132/what-columns-generally-make-good-indexes): "columns that are generally increasing and unique make a good index" so if we make them unique (see [my other comment](https://github.com/packit/packit-service/pull/1401#discussion_r829444347)), there'll be no performance issues because the values will be increasing.

Reply:
Yes, that was just a question if this is what we want. Jirka has a good point with the indexes, that's a clear solution with no speed downsides.

Reply:
I removed the reverse relationship and added uniqueness and indexing to the columns.

I also wrote into the documentation a possible explanation for an error that can occur when using the migrate-db target.
It took me a while to realize what was happening; since the service pod runs into the background through compose there is no evidence of a failure in it.

If you have any idea of a check I can do to ensure everything  is properly setup let me know.